pixeldata/src/lib.rs

```diff
@@ -0,0 +1,468 @@
+//! This crate contains the Dicom pixeldata handlers and is
+//! responsible for decoding pixeldata, such as JPEG-lossy and convert it
+//! into a [`DynamicImage`] or raw [`DecodedPixelData`].
+//!
+//! This crate is using GDCM bindings to convert
+//! different compression formats to raw pixeldata.
+//! This should become a pure Rust implementation in the future.
+//!
+//! # Examples
+//! ```no_run
+//! use dicom_object::open_file;
+//! use dicom_pixeldata::PixelDecoder;
+//! let obj = open_file("pydicom/JPEG2000.dcm").unwrap();
+//! let image = obj.decode_pixel_data().unwrap().to_dynamic_image().unwrap();
+//! image
+//!    .save("out.png")
+//!    .unwrap();
+//! ```
+
+use byteorder::{BigEndian, ByteOrder, LittleEndian};
+use dicom_core::value::Value;
+use dicom_encoding::transfer_syntax::{Endianness, TransferSyntaxIndex};
+use dicom_object::DefaultDicomObject;
+use dicom_transfer_syntax_registry::TransferSyntaxRegistry;
+use gdcm_rs::{decode_single_frame_compressed, GDCMPhotometricInterpretation, GDCMTransferSyntax};
+use image::{DynamicImage, ImageBuffer, Luma};
+use rayon::iter::{IntoParallelRefIterator, ParallelIterator};
+use snafu::OptionExt;
+use snafu::{ResultExt, Snafu};
+use std::str::FromStr;
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("Missing required element"))]
+    MissingRequiredField { source: dicom_object::Error },
+
+    #[snafu(display("Could not cast element"))]
+    CastValueError {
+        source: dicom_core::value::CastValueError,
+    },
+
+    #[snafu(display("Non supported GDCM PhotometricInterpretation: {}", pi))]
+    GDCMNonSupportedPI {
+        source: gdcm_rs::InvalidGDCMPI,
+        pi: String,
+    },
+
+    #[snafu(display("Non supported GDCM TransferSyntax: {}", ts))]
+    GDCMNonSupportedTS {
+        source: gdcm_rs::InvalidGDCMTS,
+        ts: String,
+    },
+
+    #[snafu(display("Not supported: {}", message))]
+    NotSupported { message: String },
+
+    #[snafu(display("Invalid PixelData"))]
+    InvalidPixelData,
+
+    #[snafu(display("Invalid PixelRepresentation, must be 0 or 1"))]
+    InvalidPixelRepresentation,
+
+    #[snafu(display("Invalid BitsAllocated, must be 8 or 16"))]
+    InvalidBitsAllocated,
+}
+
+pub type Result<T, E = Error> = std::result::Result<T, E>;
+
+/// Decoded pixel data
+/// NOTE: Is it possible to reuse dicom::PixelData ?
+pub struct DecodedPixelData {
```

Comment:
Hmm yes, this type would simply make the other one obsolete. It was an experiential attempt which never really served a purpose until now, so I guess we can keep this one and remove the one from the object crate.